mne_bids/dig.py

```diff
@@ -396,25 +387,38 @@ def _write_dig_bids(bids_path, raw, montage=None, acpc_aligned=False,
     if montage is None:
         montage = raw.get_montage()
     else:
-        # prevent transformation back to "head", only should be used
-        # in this specific circumstance
-        if bids_path.datatype == 'ieeg':
-            montage.remove_fiducials()
-        raw.set_montage(montage)
+        # transform back to native montage coordinates if given
+        with warnings.catch_warnings():
+            warnings.filterwarnings(action='ignore', category=RuntimeWarning,
+                                    message='.*nasion not found', module='mne')
+            raw.set_montage(montage)
+        montage_coord_frame = montage.get_positions()['coord_frame']
+        if montage_coord_frame != 'head':
+            for ch in raw.info['chs']:
+                ch['coord_frame'] = _str_to_frame[montage_coord_frame]
+            for d in raw.info['dig']:
+                d['coord_frame'] = _str_to_frame[montage_coord_frame]
 
     # get coordinate frame from digMontage
     digpoint = montage.dig[0]
     if any(digpoint['coord_frame'] != _digpoint['coord_frame']
            for _digpoint in montage.dig):
-        warn("Not all digpoints have the same coordinate frame. "
-             "Skipping electrodes.tsv writing...")
-        return
+        raise RuntimeError("Not all digpoints have the same coordinate frame.")
 
     # get the accepted mne-python coordinate frames
     coord_frame_int = int(digpoint['coord_frame'])
     mne_coord_frame = MNE_FRAME_TO_STR.get(coord_frame_int, None)
     coord_frame = MNE_TO_BIDS_FRAMES.get(mne_coord_frame, None)
 
+    if coord_frame == 'CapTrak' and \
+            bids_path.datatype == 'eeg' or bids_path.datatype == 'nirs':
```

Comment:
Better readability.

But please add parens to group the logical operations. I'm not sure which of these belong together / which one(s) should take precedence?
```suggestion
    if (
        coord_frame == 'CapTrak' and
        bids_path.datatype == 'eeg' or
        bids_path.datatype == 'nirs'
    ):
```